### PR TITLE
Cherry-pick "[SuperEditor][SuperTextField] Implement performSelector (Resolves #836, Resolves #1274) (#1339)" to stable

### LINKED ISSUE #836: [SuperEditor][SuperTextField] - Implement new performSelector IME method
We added a no-op declaration for `performSelector` when Flutter 3.3.4 was in Beta. We did this so that developers could begin to work with `super_editor` on Beta. However, we left the implementation as a no-op (https://github.com/superlistapp/super_editor/pull/835).

Once Flutter 3.3.4 moves to Stable, implement the appropriate behavior for `performSelector`.

### LINKED ISSUE #1274: Accent panel is not closed when the composing character is deleted
Steps to reproduce:
* Setup the keyboard with a language that has accents (e.g. Portuguese or German)
* Run `SuperEditor` or `SuperTextField` demos
* Press and hold letter "a" (wait for the accent panel to be shown)
* Press backspace to delete the "a"

**Expected:** the accent panel is closed
**Actual:** the accent panel remains open

https://github.com/superlistapp/super_editor/assets/31278849/6c872d94-3462-4b89-b738-511beb5afef4

### PR DESCRIPTION
**Pull request by angelosilvestre**
This PR cherry-picks "[SuperEditor][SuperTextField] Implement performSelector (Resolves #836, Resolves #1274) (#1339)" to stable.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@angelosilvestre looks like we have failing tests

**Comment by angelosilvestre** [comment]
@matthew-carroll It seems that, on Flutter stable, the selectors aren't being generated during tests. I'll need to take a look.

File: super_editor/test/super_editor/supereditor_input_keyboard_actions_test.dart [hunk 1]
```diff
@@ -222,6 +427,19 @@ void main() {
           // Press control + backspace
           await tester.pressCtlBackspace();
 
+          // TODO: remove this code after Flutter's next stable version.
```
Comment (matthew-carroll) [inline super_editor/test/super_editor/supereditor_input_keyboard_actions_test.dart:430]: It's difficult to figure out what descriptions like "next" means when added to source code. Can you please state the current Flutter version and the date.

**Comment by angelosilvestre** [comment]
@matthew-carroll The first failed tests were cause because on stable branch, `DeltaTextInputClientDecorator` wasn't forwarding the `performSelector` call to the inner client.

File: super_editor/test/super_editor/supereditor_input_keyboard_actions_test.dart (hunk 1 above)
Comment (angelosilvestre) [reply super_editor/test/super_editor/supereditor_input_keyboard_actions_test.dart:430]: Updated.

**Comment by matthew-carroll** [comment]
Given that we rebase on `stable`, rather than squash, can you squash your new commits on this branch down to a single adjustment commit?

**Comment by angelosilvestre** [comment]
@matthew-carroll Done.

**Comment by matthew-carroll** [comment]
Now it's gone too far. It looks like you've altered the original commit. I'd like to retain the original commit, with the original commit hash, and then have a single adjustment commit on top of that, which explains what it's adjusting.

**Comment by angelosilvestre** [comment]
I don't think the original commit hash will be preserved, as the parent commit is included in the hash computation. In any case, I will update it.

**Comment by matthew-carroll** [comment]
When we cherry-pick a commit, it doesn't retain the hash?

**Comment by angelosilvestre** [comment]
Nope. For example, my latest PR merged:

main branch: 4437be334b749a014c5447ebb74dcacf48852353
stable branch: 1da7c9556c511b8e22d55d5ff787219169782abd

When we cherry-pick a commit, the parent commit changes (the latest commit of the branch) and the parent commit is considered to compute the final hash.

**Comment by matthew-carroll** [comment]
That's annoying. Makes it more difficult to maintain two parallel branches.

**Comment by angelosilvestre** [comment]
@matthew-carroll Updated

**Pull request merged**